src/main/java/mx/erickb/shipping/util/RouteUtils.java

```diff
@@ -0,0 +1,80 @@
+package mx.erickb.shipping.util;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import mx.erickb.shipping.exception.InvalidResponseException;
+import mx.erickb.shipping.model.Route;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Component
+public class RouteUtils {
+
+    private final Logger logger = LoggerFactory.getLogger(RouteUtils.class);
+    private final ObjectMapper mapper;
+
+    public RouteUtils(ObjectMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    public List<Route> parseRoutesResponse(String routesResponse) {
+        try {
+            Route[] routes = mapper.readValue(routesResponse, Route[].class);
+            return Arrays.asList(routes);
+        } catch (JsonProcessingException e) {
+            logger.error("Error when processing server response (" + e.getMessage() + ")");
```

Comment:
change this logger to 
```
logger.error("Error when processing server response", e);
```
so the logger can log the stack trace, very useful in troubleshooting